aries/misc/legacy/libvcx_core/src/api_vcx/api_global/wallet.rs

```diff
@@ -208,21 +259,44 @@ pub async fn wallet_delete_wallet_record_tags(
     tags_json: &str,
 ) -> LibvcxResult<()> {
     let wallet = get_main_wallet()?;
-    map_ariesvcx_core_result(wallet.delete_wallet_record_tags(xtype, id, tags_json).await)
+    let tags: HashMap<String, String> = serde_json::from_str(tags_json)?;
+
+    let record = wallet.get_record(xtype, id).await?;
+
+    let mut found_tags = IndyTags::from_entry_tags(record.tags().clone()).to_inner();
+    for key in tags.keys() {
+        found_tags.remove(key);
+    }
+
+    map_ariesvcx_core_result(
+        wallet
+            .update_record_tags(xtype, id, IndyTags::new(found_tags).to_entry_tags())
+            .await,
+    )
 }
 
 pub async fn wallet_get_wallet_record(
     xtype: &str,
     id: &str,
-    options: &str,
+    _options: &str,
 ) -> LibvcxResult<String> {
     let wallet = get_main_wallet()?;
-    map_ariesvcx_core_result(wallet.get_wallet_record(xtype, id, options).await)
+
+    map_ariesvcx_result(
+        wallet
+            .get_record(xtype, id)
+            .map(|res| {
+                let wallet_record = WalletRecord::from_record(res?)?;
+
+                Ok(serde_json::to_string(&wallet_record)?)
+            })
+            .await,
+    )
```

Comment:
Why doesn't `WalletRecord` implement `Serialize`? This entire function can be a single line:
```rust
serde_json::to_string(&get_main_wallet()?.get_record(id, xtype).await?).map_err(Into::into)
```

Reply:
Because`Record` would need to implement `Serialize` exactly the same way as `WalletRecord`.

Reply:
Sorry, I missed the point of the function. It (and others in this module) can be simplified regardless, although no point, this is code to be removed as per location. `Record` could definitely implement `Serialize`, but also no need.